src/mbgl/text/placement.cpp

```diff
@@ -329,6 +331,15 @@ void Placement::placeBucket(
                                                                     allowOverlap,
                                                                     pitchWithMap,
                                                                     params.showCollisionBoxes, avoidEdges, collisionGroup.second, textBoxes);
+
+                        if (doVariableIconPlacement) {
+                            auto placedIconFeature = collisionIndex.placeFeature(iconCollisionFeature, shift,
```

Comment:
do we still need to call `placedFeature = collisionIndex.placeFeature(..` above? wouldn't just one `placeFeature()` call suffice?

Reply:
In the case there is negative icon padding (or an icon offset?) the text box might be larger than the icon box. Placing both is the clearest way to handle the possible differences